### PR TITLE
Add support for BlackBerry Passport

### PR DESCRIPTION
**Pull request by BotchedRPR**
Hi all.
This PR adds support for the BlackBerry Passport (wolverine). 
Was tested on my personal device (European distribution).

Best regards
Igor

### CODE REVIEW

File: lk2nd/device/dts/msm8974/msm8974-blackberry-wolverine.dts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <skeleton64.dtsi>
+#include <lk2nd.dtsi>
+
+/ {
+	qcom,msm-id = <217 4034135218 0x10001>;
```
Comment (TravMurav) [inline lk2nd/device/dts/msm8974/msm8974-blackberry-wolverine.dts:7]:
```suggestion
	qcom,msm-id = <QCOM_ID_MSM8974PRO_AA 0xf07404b2 0x10001>;
```

and I'd also

**Review by TravMurav — COMMENTED** [review]
This is quite impressive, didn't know one can run alternate software on BB devices. Could you help me understand how you got lk2nd installed? Perhaps making a photo with lk2nd menu on the screen would be cool to see as well :)

**Comment by BotchedRPR** [comment]
> This is quite impressive, didn't know one can run alternate software on BB devices. Could you help me understand how you got lk2nd installed? Perhaps making a photo with lk2nd menu on the screen would be cool to see as well :)

Thanks for the review!
Well, the process isn't quite simple. You need to replace the eMMC of the device with one that has been prepared with leaked bootloader(s) from some Android prototypes. Also, if you use a secureboot exploit, it will be off due to an exploit in the Krait cores, so lk1st will be possible too :)

This process and tools were made by balika011, everything needed is here: https://github.com/BBAndroids

And here is a picture:
photo_2024-12-31_07-44-34

File: lk2nd/device/dts/msm8974/msm8974-blackberry-wolverine.dts (hunk 1 above)
Comment (BotchedRPR) [reply lk2nd/device/dts/msm8974/msm8974-blackberry-wolverine.dts]: Thanks! Resolved.

**Comment by TravMurav** [comment]
Shame the installation process isn't at all trivial, but regardless good job! (a bit sad that the font selection logic broke for this screen but no big deal haha)

Thanks!

**Pull request merged**